### PR TITLE
refactor: use unified repository plugin handler

### PR DESCRIPTION
**Pull request by a-cordier**
see https://github.com/gravitee-io/issues/issues/3145

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/3145-unified-repository-plugin-handler/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-szkjrffgqr.chromatic.com)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -70,6 +70,7 @@
         <gravitee-node.version>1.24.2</gravitee-node.version>
         <gravitee-notifier-api.version>1.4.1</gravitee-notifier-api.version>
         <gravitee-plugin.version>1.24.1</gravitee-plugin.version>
+        <gravitee-repository-api.version>1.2.0</gravitee-repository-api.version>
```
Comment (phiz71) [inline pom.xml]:
```suggestion
        <gravitee-platform-repository-api.version>1.2.0</gravitee-platform-repository-api.version>
```

File: pom.xml [hunk 2]
```diff
@@ -309,6 +310,18 @@
...
+            <dependency>
+                <groupId>io.gravitee.plugin</groupId>
+                <artifactId>gravitee-plugin-repository</artifactId>
+                <version>${gravitee-plugin.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>io.gravitee.platform</groupId>
+                <artifactId>gravitee-platform-repository-api</artifactId>
+                <version>${gravitee-repository-api.version}</version>
```
Comment (phiz71) [inline pom.xml]:
```suggestion
                <version>${gravitee-platform-repository-api.version}</version>
```

File: gravitee-apim-repository/gravitee-apim-repository-hazelcast/pom.xml [hunk 3]
```diff
@@ -39,6 +39,11 @@
 			<scope>provided</scope>
 		</dependency>
 
+		<dependency>
+			<groupId>io.gravitee.platform</groupId>
+			<artifactId>gravitee-platform-repository-api</artifactId>
+		</dependency>
+
```
Comment (jhaeyaert) [inline gravitee-apim-repository/gravitee-apim-repository-hazelcast/pom.xml]: Globally speaking when used in a plugin, should this dependency be in scope provided as is already provided by the container (gateway or management) ?

**Review by jhaeyaert — APPROVED** [review]
I didn't run it locally but browsed all the files and all looks good to me. I've also compared with the work done on AM  (2 years ago 😅) and everything appears to be ok 👍 

Just one comment regarding a dependency scope.

Well done 🤩

**Pull request merged**